src/runtime_src/core/edge/drm/zocl/zocl_ioctl.c

```diff
@@ -664,11 +585,42 @@ zocl_read_axlf_ioctl(struct drm_device *ddev, void *data, struct drm_file *filp)
 		goto out0;
 	}
 
-	if (axlf_obj->za_flags & DRM_ZOCL_AXLF_FLAGS_PDI_LOAD) {
-		/* If PDI section is available, load PDI */
-		size = zocl_read_sect(PDI, &pdi_buffer, axlf, xclbin);
+	/* For PR support platform, device-tree has configured addr */
+	if (zdev->pr_isolation_addr &&
+	    (axlf_obj->za_flags & DRM_ZOCL_AXLF_BITSTREAM)) {
+		size = zocl_read_sect(BITSTREAM, &section_buffer,
+		    axlf, xclbin);
+		if (size > 0) {
+			ret = zocl_load_bitstream(zdev, section_buffer, size);
+			vfree(section_buffer);
+			section_buffer = NULL;
+			if (ret)
+				goto out0;
+		} else if (size < 0) {
```

Comment:
This is a little bit weird we leave size == 0, is it what you mean? what is special for 0 size that we don't treat is as an error? And your size is an uint64, it will not be less than 0.

Another suggestion is, we seem to duplicate code here. Can we make a function and all three za_flags call the same function?